### <p style="font-family:JetBrains Mono; font-weight:bold; letter-spacing: 2px; color:#294B8E; font-size:140%; text-align:left;padding: 0px; border-bottom: 3px solid #294B8E">Исследование интернет-компании</p>

<a id="1.2"></a>
<div style="padding:20px;
            color:white;
            margin:10;
            font-size:170%;
            text-align:left;
            display:fill;
            border-radius:5px;
            background-color:#294B8E;
            overflow:hidden;
            font-weight:700"><span style='color:#CDA63A'>|</span> Содержание</div>

<div class="toc"><ul class="toc-item"><li><span><a href="#Загрузка-библиотек" data-toc-modified-id="Загрузка-библиотек-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Загрузка библиотек</a></span></li>
    <li><span><a href="#EDA" data-toc-modified-id="EDA-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>EDA</a></span></li>
    <li><span><a href="#Предобратка-данных" data-toc-modified-id="Предобратка-данных-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Предобратка данных</a></span></li>
    <li><span><a href="#a. Прибыльный-рекламный-канал" data-toc-modified-id="a.-Прибыльный-рекламный-канал-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>a. Прибыльный рекламный канал</a></span></li>
    <li><span><a href="#b.-Исследование-среднего-чека-транзакции" data-toc-modified-id="b.-Исследование-среднего-чека-транзакции-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>b. Исследование среднего чека транзакции</a></span></li>
     <li><span><a href="#c.-Исследование-CR" data-toc-modified-id="c.-Исследование-CR-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>c. Исследование CR</a></span></li>
     <li><span><a href="#d.-Прогнозирование-объема-дохода" data-toc-modified-id="d.-Прогнозирование-объема-дохода-7"><span class="toc-item-num">7&nbsp;&nbsp;</span>d. Прогнозирование объема дохода</a></span></li>

<a id="Загрузка-библиотек"></a>
<div style="padding:20px;
            color:white;
            margin:10;
            font-size:170%;
            text-align:left;
            display:fill;
            border-radius:5px;
            background-color:#294B8E;
            overflow:hidden;
            font-weight:700"><span style='color:#CDA63A'>|</span> Загрузка библиотек</div>

In [1]:
import pandas as pd

<a id="EDA"></a>
<div style="padding:20px;
            color:white;
            margin:10;
            font-size:170%;
            text-align:left;
            display:fill;
            border-radius:5px;
            background-color:#294B8E;
            overflow:hidden;
            font-weight:700"><span style='color:#CDA63A'>|</span> EDA</div>

In [2]:
summer = pd.read_csv('summer_camp_data.csv')

In [3]:
summer

date    source   medium delivery_available device_type  \
0      2020-05-11    google  organic      Не определено     Десктоп   
1      2020-05-11    yandex      cpc      Не определено      Мобайл   
2      2020-05-11    google      cpc      Не определено      Мобайл   
3      2020-05-11    google      cpc      Не определено     Десктоп   
4      2020-05-11    yandex  organic      Не определено     Десктоп   
...           ...       ...      ...                ...         ...   
52716  2020-06-21  (direct)   (none)      Доставка есть      Мобайл   
52717  2020-08-09  co-promo    email      Доставка есть      Мобайл   
52718  2020-09-22    google      cpc      Доставка есть      Мобайл   
52719  2020-08-25  (direct)   (none)      Не определено      Мобайл   
52720  2020-03-19   admitad      cpa      Не определено     Десктоп   

      promo_activated filter_used  pageviews  visits  productClick  addToCart  \
0                  no          no       3120    1233          5240       1048   
1                 yes          no       3302     544          9930       1984   
2                  no          no       2970    1450          5460       1090   
3                  no          no       1875     854          4250        848   
4                  no          no       2159    1000          4110        824   
...               ...         ...        ...     ...           ...        ...   
52716             yes         yes         65       4           250         50   
52717              no          no         43       2           250         50   
52718             yes         yes        153       3           250         50   
52719             yes         yes        219       8           250         50   
52720             yes         yes         83       5           250         50   

       checkout  transactions   revenue  
0           525            90   86649.0  
1          1416           217  244478.0  
2           599           100  105150.0  
3           407            71   79003.0  
4           351            62   61861.0  
...         ...           ...       ...  
52716        58             4    4821.0  
52717        23             1    7348.0  
52718        58             1     958.0  
52719        45             1     828.0  
52720        25             4    4078.0  

[52721 rows x 14 columns]

In [4]:
summer.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 52721 entries, 0 to 52720
Data columns (total 14 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   date                52721 non-null  object 
 1   source              52721 non-null  object 
 2   medium              52721 non-null  object 
 3   delivery_available  52721 non-null  object 
 4   device_type         52721 non-null  object 
 5   promo_activated     52721 non-null  object 
 6   filter_used         52721 non-null  object 
 7   pageviews           52721 non-null  int64  
 8   visits              52721 non-null  int64  
 9   productClick        52721 non-null  int64  
 10  addToCart           52721 non-null  int64  
 11  checkout            52721 non-null  int64  
 12  transactions        52721 non-null  int64  
 13  revenue             52721 non-null  float64
dtypes: float64(1), int64(6), object(7)
memory usage: 5.6+ MB


In [5]:
summer.duplicated().sum()

0

<div class="alert alert-block alert-info" style="background-color:#FFFAF0;
                                                 color:black;
                                                 border-color:#DEB887;
                                                 border-radius:5px;
                                                 width:50%;
                                                 margin: auto;
                                                 text-align: left;">

<b>✍ Вывод</b>
<br>В данных отсутствуют дубликаты и пропуски. Необходимо поменять тип столбца "date". Обнаружено некорректное название столбцов. 
        
</div>

<a id="Предобратка-данных"></a>
<div style="padding:20px;
            color:white;
            margin:10;
            font-size:170%;
            text-align:left;
            display:fill;
            border-radius:5px;
            background-color:#294B8E;
            overflow:hidden;
            font-weight:700"><span style='color:#CDA63A'>|</span> Предобработка данных</div>

In [6]:
summer = summer.rename(columns={'productClick': 'product_click',
                               'addToCart': 'add_to_cart',
                               'checkout': 'check_out'})

In [7]:
#поменяем тип столбца 'Date_created'
summer['date'] = pd.to_datetime(summer['date'], format="%Y-%m-%dT%H:%M:%S")

In [8]:
summer.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 52721 entries, 0 to 52720
Data columns (total 14 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   date                52721 non-null  datetime64[ns]
 1   source              52721 non-null  object        
 2   medium              52721 non-null  object        
 3   delivery_available  52721 non-null  object        
 4   device_type         52721 non-null  object        
 5   promo_activated     52721 non-null  object        
 6   filter_used         52721 non-null  object        
 7   pageviews           52721 non-null  int64         
 8   visits              52721 non-null  int64         
 9   product_click       52721 non-null  int64         
 10  add_to_cart         52721 non-null  int64         
 11  check_out           52721 non-null  int64         
 12  transactions        52721 non-null  int64         
 13  revenue             52721 non-null  float64   

<a id="a.-Прибыльный-рекламный-канал"></a>
<div style="padding:20px;
            color:white;
            margin:10;
            font-size:170%;
            text-align:left;
            display:fill;
            border-radius:5px;
            background-color:#294B8E;
            overflow:hidden;
            font-weight:700"><span style='color:#CDA63A'>|</span> a. Прибыльный рекламный канал</div>

In [9]:
sum_medium = 0
for i in summer['medium'].unique():
    sum_medium = summer[summer['medium'] == i]['revenue'].sum()
    print('Cумма дохода ', i, ' = ', '{0:,}'.format(sum_medium).replace(',', '.')
)

Cумма дохода  organic  =  498.455.154.0
Cумма дохода  cpc  =  444.837.608.0
Cумма дохода  (none)  =  238.355.148.25
Cумма дохода  cpa  =  81.834.240.0
Cумма дохода  email  =  11.000.859.0
Cумма дохода  social  =  913.626.0


<div class="alert alert-block alert-info" style="background-color:#FFFAF0;
                                                 color:black;
                                                 border-color:#DEB887;
                                                 border-radius:5px;
                                                 width:50%;
                                                 margin: auto;
                                                 text-align: left;">

<b>✍ Вывод</b>
<br>Самый прибыльный рекламный ролик за все время organic, его доход составляет примерно 498 млн.    
</div>

<a id="b.-Исследование-среднего-чека-транзакции"></a>
<div style="padding:20px;
            color:white;
            margin:10;
            font-size:170%;
            text-align:left;
            display:fill;
            border-radius:5px;
            background-color:#294B8E;
            overflow:hidden;
            font-weight:700"><span style='color:#CDA63A'>|</span> b. Исследование среднего чека транзакции</div>

In [10]:
#cредний чек транзакции для пользователей, покупающих с промокодом
round(summer[summer['promo_activated'] == 'yes']['transactions'].mean(),2)

21.7

In [11]:
#cредний чек транзакции для пользователей, покупающих без промокода
round(summer[summer['promo_activated'] == 'no']['transactions'].mean(),2)

18.61

<div class="alert alert-block alert-info" style="background-color:#FFFAF0;
                                                 color:black;
                                                 border-color:#DEB887;
                                                 border-radius:5px;
                                                 width:50%;
                                                 margin: auto;
                                                 text-align: left;">

<b>✍ Вывод</b>
<br> Cредний чек транзакции для пользователей, покупающих с промокодом в 1,16 раз больше, чем для пользователей, покупающих без промокодом.
</div>

<a id="c.-Исследование-CR"></a>
<div style="padding:20px;
            color:white;
            margin:10;
            font-size:170%;
            text-align:left;
            display:fill;
            border-radius:5px;
            background-color:#294B8E;
            overflow:hidden;
            font-weight:700"><span style='color:#CDA63A'>|</span> c. Исследование CR</div>

In [12]:
summer['weekday'] = summer['date'].dt.weekday

In [13]:
count_transactions = summer['transactions'].count()

In [14]:
count_transactions_workday = summer[(summer['weekday'] == 0) |
                                         (summer['weekday'] == 1) | 
                                         (summer['weekday'] == 2) |
                                         (summer['weekday'] == 3) |
                                         (summer['weekday'] == 4)]['transactions'].count()

In [15]:
count_transactions_weekday = summer[(summer['weekday'] == 5) |
                                         (summer['weekday'] == 6)]['transactions'].count()

In [16]:
CR_workday = round(count_transactions_workday / 52721 * 100, 2)
CR_workday

70.64

In [17]:
CR_weekday = round(count_transactions_weekday /52721 * 100, 2)
CR_weekday

29.36

<div class="alert alert-block alert-info" style="background-color:#FFFAF0;
                                                 color:black;
                                                 border-color:#DEB887;
                                                 border-radius:5px;
                                                 width:50%;
                                                 margin: auto;
                                                 text-align: left;">

<b>✍ Вывод</b>
<br> CR (коэффициент конверсии в транзакцию) в будние дни выше в 2,4 раза CR в выходные дни.
</div>

<a id="d.-Прогнозирование-объема-дохода"></a>
<div style="padding:20px;
            color:white;
            margin:10;
            font-size:170%;
            text-align:left;
            display:fill;
            border-radius:5px;
            background-color:#294B8E;
            overflow:hidden;
            font-weight:700"><span style='color:#CDA63A'>|</span> d. Прогнозирование объема дохода</div>

<div class="alert alert-block alert-info" style="background-color:#FFFAF0;
                                                 color:black;
                                                 border-color:#DEB887;
                                                 border-radius:5px;
                                                 width:50%;
                                                 margin: auto;
                                                 text-align: left;">
<b>✍ Вывод</b>
<br> 1. Изучение исторических данных за прошлый год (2019) пользователей, приведенных на сайт контекстной рекламой.
<br> 2. Заполнение нового датасета с новыми датами используя анализ исторических данных  
<br> 3. Предобработка данных
<br> 4. Деление исторических данных на обучающую и валидационную выборки. Новый датасет с новыми датами, заполненный с помощью историчесикз данных, использовать как тестовую выборку.   
<br> 5. Прогнозирование объема дохода для новых дат используя обученные модели (например, Catboost, LightGBM, XGBoost, Random Forest, Desicion Tree, Linear Regression, Lasso regression, Ridge regression) на исторических данных.
</div>